book/src/out-of-order-blocks.md

```diff
@@ -0,0 +1,125 @@
+# Out Of Order Blocks
+
+The [deterministic leader schedule](../tree/master/book/src/leader-rotation.md)
+creates a fixed schedule for block production.  The main problems with this
+approach are as follows:
+
+* The leader schedule has to balance both performance and liveness requirements.
+The epoch needs to be long enough to guarantee liveness.
+
+* The cluster has to wait for unavailable nodes.
+
+* The cluster's liveness depends on the availability of scheduled nodes.
+
+* Attackers know in advance all of the leaders that propose a block.
+
+The general problems of DPOS — that a cluster requires an interactive protocol
+for account registration, delegation, and voting that a set of leaders can
+censor — is not addressed in this design.
+
+## Random Leaders
+
+Leaders can generate a *golden ticket* to propose a block.
+
+* *N* number of bits of a leader's vote signature that must match a PoH hash.
```

Comment:
I am a bit confused here. Ticks are also PoH hashes, right. So what do you mean by between ticks?

Reply:
@pgarg66 we are looking for a magick hash anywhere in PoH, not necessarily at a tick boundary.

Reply:
So, is Golden ticket generated as a special hash, or it happens to be any hash in PoH that matches N bits of node's vote signature? Also, do we mean vote signature of voting key?

Reply:
any hash in PoH that matches the N bits.

Reply:
That could mean multiple golden tickets in a slot, matching different nodes.. How is contention resolved (maybe how many bits match)?

Reply:
@pgarg66 earliest based on PoH hash count is what the network will forward transactions to.  But either block can win, since they are concurrent with the deterministic leader schedule.

Reply:
Does the golden ticket out-prioritize the deterministic leader?

Reply:
@carllin regular fork selection rules apply

Reply:
how does this impact the transaction forwarding?